src/components/Buttons/ButtonsTask/ButtonsTask.jsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button } from '../Button/Button';
+import { BUTTONS_NAMES, BUTTONS_TYPES } from '../../../shared/constants';
+import { getTaskData, removeTask, getAllTasks } from '../../../services/tasks-services';
+import { DeleteForm } from '../../Forms/DeleteForm/DeleteForm';
+import { MyModal } from '../../Common/Modal/Modal';
+import { getAllUsers } from '../../../services/users-services ';
+import { CreateTaskForm } from '../../Forms/CreateTaskForm/CreateTaskForm';
+
+export class ButtonsTask extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isDeleteModalOpen: false,
+      isEditModalOpen: false,
+      taskData: {},
+      allUsers: [],
+    };
+  }
+
+  async componentDidMount() {
+    const allUsers = await getAllUsers();
+    if (allUsers) {
+      this.setState({ allUsers });
+    } else {
+      this.toggleError();
+    }
+  }
+
+  async getTasksData() {
+    const { id, toggleError } = this.props;
+    const taskData = await getTaskData(id);
+    if (taskData) {
+      this.setState({ taskData });
+    } else {
+      toggleError();
+    }
+  }
+
+  deleteTaskHandler = async () => {
+    const { id, setTasksHandler, toggleError } = this.props;
+    const isDeleted = await removeTask(id);
+    if (isDeleted) {
+      const tasks = await getAllTasks();
+      setTasksHandler(tasks);
```

Comment:
I don't really understand your code here. Looks like what you do here is not actually right. You should use ComponentDidUpdate lifecycle hook to update tasks after deletion. You should do the same everywhere and you don't need this strange `isDeleted` flag in this case.

Reply:
Yes, I described it above.